ios/MVVM_Testing.md

```diff
@@ -0,0 +1,198 @@
+## MVVM Testing Walkthrough
+
+To better understand testing in an MVVM architected app we will walkthrough a simple app and highlight testing approaches for the related view models.
+
+This post will rely on a basic music app that includes a single screen for searching for albums from a web API. Each album in the search result will be displayed as a table view row in a SearchViewController. The table view displays each album as an AlbumSearchResultTableViewCell. The AlbumSearchResultTableViewCell displays an album title in a single label and the artist name, album release year on a second label and the album cover art.
+
+In terms of view models, the SearchViewController is backed by a SearchViewModel and each AlbumSearchResultTableViewCell is backed by a AlbumSearchResultCellViewModel.
+
+Provided below is the album model and the relevant view models.
+
+```
+struct Album {
+    let title: String
+    let artist: String
+    let releaseDate: Date
+    let imageURL: URL
+
+    init(title: String, artist: String, releaseDate: Date, imageURL: URL) {
+        self.title = title
+        self.artist = artist
+        self.releaseDate = releaseDate
+        self.imageURL = imageURL
+    }
+}
+```
+
+```
+class SearchViewModel {
+
+    private let networkingLayer: NetworkingLayer
+
+    var albumViewModels: [AlbumSearchResultCellViewModel] = []
+
+    init(networkingLayer: NetworkingLayer = NetworkingLayer()) {
+        self.networkingLayer = networkingLayer
+    }
+
+    func albumViewModelAt(index: Int) -> AlbumSearchResultCellViewModel {
+        return albumViewModels[index]
+    }
+
+    func searchFor(searchTerm: String, completionHandler:@escaping () -> ()) {
+        networkingLayer.searchFor(searchTerm: searchTerm) { [weak self] (albums) in
+
+        self?.albumViewModels = albums.map( {AlbumSearchResultCellViewModel(album: $0)} )
+
+
```

Comment:
Clean up some of this whitespace.  Also the indentation looks wrong for `self?.albumViewModels =...`